meson.build

```diff
@@ -0,0 +1,1788 @@
+    flags += ['-isystem', '-I', '-DIN_WINPTHREAD']
+    # Note: Adding this include causes POSIX_TIMERS to be defined for
+    # unclear reasons (some confusion with <pthread.h> probably).
+    # Hack around it by using HAVE_WIN32_INTERNAL_PTHREADS.
+    includedir += include_directories('osdep/win32/include')
+    sources += files('osdep/win32/pthread.c')
+endif
+
+pthread_debug = get_option('pthread-debug').require(
+    win32_pthreads.disabled(),
+    error_message: 'win32-internal-pthreads was found!'
+)
+if pthread_debug.allowed()
+    features += 'pthread-debug'
+    flags += '-DMP_PTHREAD_DEBUG'
+endif
+
+add_project_arguments(flags, language: 'c')
+add_project_link_arguments(link_flags, language: ['c', 'objc'])
+
+
+# osdep
+cocoa = dependency('appleframeworks', modules: ['Cocoa', 'IOKit', 'QuartzCore'],
+                   required: get_option('cocoa'))
+if cocoa.found()
+    dependencies += cocoa
+    features += 'cocoa'
+    sources += files('osdep/macosx_application.m',
+                     'osdep/macosx_events.m',
+                     'osdep/macosx_menubar.m',
+                     'osdep/main-fn-cocoa.c',
+                     'osdep/path-macosx.m',
+                     'video/out/cocoa_common.m',
+                     'video/out/cocoa/events_view.m',
+                     'video/out/cocoa/video_view.m',
+                     'video/out/cocoa/window.m')
+endif
+
+if posix
+    sources += files('input/ipc-unix.c',
+                     'osdep/path-unix.c',
+                     'osdep/polldev.c',
+                     'osdep/subprocess-posix.c',
+                     'osdep/terminal-unix.c',
+                     'sub/filter_regex.c')
+endif
+
+if posix and not cocoa.found()
+    sources += files('osdep/main-fn-unix.c')
+endif
+
+if darwin
+    sources += files('osdep/timer-darwin.c')
+endif
+
+if posix and not darwin
+    sources += files('osdep/timer-linux.c')
+endif
+
+cd_devices = {
+    'windows': 'D:',
+    'cygwin': 'D:',
+    'darwin': '/dev/disk1',
+    'freebsd': '/dev/cd0',
+    'openbsd': '/dev/rcd0c',
+    'linux': '/dev/sr0',
+}
+if host_machine.system() in cd_devices
+    cd_device = cd_devices[host_machine.system()]
+else
+    cd_device = '/dev/cdrom'
+endif
+
+dvd_devices = {
+    'windows': 'D:',
+    'cygwin': 'D:',
+    'darwin': '/dev/diskN',
+    'freebsd': '/dev/cd0',
+    'openbsd': '/dev/rcd0c',
+    'linux': '/dev/sr0',
+}
+if host_machine.system() in cd_devices
+    dvd_device = dvd_devices[host_machine.system()]
+else
+    dvd_device = '/dev/dvd'
+endif
+
+android = host_machine.system() == 'android'
+if android
+    features += 'android'
+    sources += files('audio/out/ao_audiotrack.c',
+                     'video/out/android_common.c',
+                     'video/out/vo_mediacodec_embed.c')
+endif
+
+uwp = {
+    'name': 'uwp',
+    'deps': cc.find_library('windowsapp', required: get_option('uwp')),
+    'use': false,
+}
+if uwp['deps'].found() and not get_option('cplayer')
```

Comment:
```meson
uwp_opt = get_option('uwp').require(not get_option('cplayer'), error_message: 'cplayer is not false!')
uwp = cc.find_library('windowsapp', required: uwp_opt)
if uwp.found()
    dependencies += uwp
    features += 'uwp'
    sources += files('osdep/path-uwp.c')
endif
```

Reply:
Also hmm, the current dependencies += uwp, is adding that whole dict to dependencies, rather than `uwp['deps']`.

Reply:
This is one of those options I'm not sure anyone actually uses but uh yeah let's unbreak this.